cardano-cli/src/Cardano/CLI/Run/Hash.hs

```diff
@@ -36,21 +50,26 @@ runHashAnchorDataCmd
   :: ()
   => Cmd.HashAnchorDataCmdArgs
   -> ExceptT HashCmdError IO ()
-runHashAnchorDataCmd Cmd.HashAnchorDataCmdArgs{toHash, mOutFile} =
-  case toHash of
-    Cmd.AnchorDataHashSourceBinaryFile fp -> do
-      let path = unFile fp
-      bytes <- handleIOExceptT (HashReadFileError path) $ BS.readFile path
-      let hash = L.hashAnchorData $ L.AnchorData bytes
-      writeHash hash
-    Cmd.AnchorDataHashSourceTextFile fp -> do
-      let path = unFile fp
-      text <- handleIOExceptT (HashReadFileError path) $ Text.readFile path
-      let hash = L.hashAnchorData $ L.AnchorData $ Text.encodeUtf8 text
-      writeHash hash
-    Cmd.AnchorDataHashSourceText text -> do
-      let hash = L.hashAnchorData $ L.AnchorData $ Text.encodeUtf8 text
-      writeHash hash
+runHashAnchorDataCmd Cmd.HashAnchorDataCmdArgs{toHash, mExpectedHash, mOutFile} = do
+  anchorData <-
+    L.AnchorData <$> case toHash of
+      Cmd.AnchorDataHashSourceBinaryFile fp -> do
+        let path = unFile fp
+        handleIOExceptT (HashReadFileError path) $ BS.readFile path
+      Cmd.AnchorDataHashSourceTextFile fp -> do
+        let path = unFile fp
+        text <- handleIOExceptT (HashReadFileError path) $ Text.readFile path
+        return $ Text.encodeUtf8 text
+      Cmd.AnchorDataHashSourceText text -> return $ Text.encodeUtf8 text
+      Cmd.AnchorDataHashSourceURL urlText ->
+        getByteStringFromURL urlText
+  let hash = L.hashAnchorData anchorData
+  case mExpectedHash of
+    Just expectedHash
+      | hash /= expectedHash ->
+          left $ HashMismatchedHashError expectedHash hash
+      | otherwise -> liftIO $ putStrLn "Hashes match!"
+    Nothing -> writeHash hash
```

Comment:
This code bit is surprising a little: specifying the hash in the CLI argument, makes the command ignore `--out-file` argument. I agree that's redundant here, but for the sake of consistency, I would still add writing output file even when the hash is specified as a command line argument.

Reply:
I have now changed it for outputting only in case of `--out-file` and not in case of `stdout` (https://github.com/IntersectMBO/cardano-cli/pull/895/commits/dda90abab7aa4ac35b289131512c4c07f5d8dad2)
I the case of `stdout` I am guessing it would be confusing to mix both results

Reply:
Still will be mixed usually, and it is not an error. Not sure. Another option is to print the message to a file instead of the hash when `--out-file` is used. Or to make them mutually exclusive

Reply:
Well stderr isn't reserved only for errors, but is usually used for any kind of diagnostics. I guess ignoring `--out-file` would work too, but if we print a warning here.

@CarlosLopezDeLara thoughts on the UX here?

Reply:
I have tried making them mutually exclusive in: https://github.com/IntersectMBO/cardano-cli/pull/895/commits/cabb16af758f58e03844918855725f2e88c26b6b
I think that works quite well, maybe the problem was trying to do two conflicting things at the same time

Reply:
Nice, I like it!